pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -625,13 +634,19 @@ ssize_t lwip_sock_send(struct netconn *conn, const void *data, size_t len,
         netbuf_delete(buf);
         return -ENOTCONN;
     }
-    res = len;  /* set for non-TCP calls */
+    res = payload_len;  /* set for non-TCP calls */
     if (remote != NULL) {
         err = netconn_sendto(tmp, buf, &remote_addr, remote_port);
     }
 #if LWIP_TCP
     else if (tmp->type & NETCONN_TCP) {
-        err = netconn_write_partly(tmp, data, len, 0, (size_t *)(&res));
+        while (snips) {
+            err = netconn_write_partly(tmp, snips->data, snips->len, 0, (size_t *)(&res));
```

Comment:
Can we not use the buf created above here? And just do one send?

Reply:
Yea I was wondering why `buf` was not used at all in the TCP case.
I suppose we can do 

```C
netconn_write_partly(tmp, buf->p->payload, buf->p->len, 0, (size_t *)(&res));
```

here